File: solutions/golang/stackOverFlow/comment.go
package stackoverflow

import "time"

type Comment struct {
	ID           int
	Content      string
	Author       *User
	CreationDate time.Time
}

func NewComment(author *User, content string) *Comment {
	return &Comment{
		ID:           generateID(),
		Author:       author,
		Content:      content,
		CreationDate: time.Now(),
	}
}

File: solutions/csharp/stackoverflow/Comment.cs
using System;

namespace StackOverflow
{
    public class Comment
    {
        public int Id { get; }
        public string Content { get; }
        public User Author { get; }
        public DateTime CreationDate { get; }

        public Comment(User author, string content)
        {
            Id = GenerateId();
            Author = author;
            Content = content;
            CreationDate = DateTime.Now;
        }

        private int GenerateId()
        {
            return (int)(DateTime.Now.Ticks % int.MaxValue);
        }
    }
}